### PR TITLE
feat: add button

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

File: src/inputs/Button/Button.tsx [hunk 1]
```diff
@@ -1,44 +1,41 @@
...
+	const loadingClass = isLoading ? (variant === 'primary' ? styles.loadingPrimary : styles.loadingSecondary) : ''
 
-  const dynamicClasses = [
-    styles.template,
-    variantClass,
-    getDisabledClassForSpecificVariant(),
-  ].join(" ");
+	const allClasses = [globalButtonClass, variantClass, disabledClass, loadingClass].join(' ')
+	return (
+		<button className={allClasses} type={type} disabled={disabled}>
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:28]: **nitpick:** As you are more experienced dev now, I have to stick to missing formatting. Look how hard to read that when you don't use any formatter in your code

File: src/inputs/Button/Button.stories.tsx [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
+	isLoading: false,
+}
 
-Button.args = {
-  disabled: false,
-  variant: "primary",
-};
+export const Icon: Story = args => (
+	<Button isLoading={false} variant={'primary'} {...args} onClick={() => console.log('clicked')}>
+		<div className={styles.button__container}>
```
Comment (Sumick) [inline src/inputs/Button/Button.stories.tsx:24]: **issue:** It definitely should not look like that. First of all, `icon` should be a prop. You Button with icon should look like following: 

```javascript
<Button isLoading={false} icon={<Plus />} variant={'primary'} {...args} onClick={() => console.log('clicked')}/>
```

File: src/inputs/Button/Button.tsx (hunk 1 above)
Comment (Sumick) [reply src/inputs/Button/Button.tsx:28]: oh, this is open source - so it should format your code here. Please add prettier formatting to `husky` file.

**Review by Sumick — CHANGES_REQUESTED** [review]
Few things to fix

File: src/inputs/Button/Button.tsx [hunk 3]
```diff
@@ -1,44 +1,52 @@
...
+	type = 'button',
+	disabled = false,
+	isLoading = false,
+	variant = 'primary',
+	children,
+}) => {
+	console.log(calculateDefaultDimensions(16, 50));
+	const globalButtonClass = styles.buttons
+	const variantClass = variant === 'primary' ? styles.primary : styles.secondary
+	const disabledClass = disabled ? (variant === 'primary' ? styles.disabledPrimary : styles.disabledSecondary) : ''
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:27]: **issue:** We are using `cx` for joining classes - check how its done in `TextField`

File: src/inputs/Button/Button.tsx [hunk 4]
```diff
@@ -1,44 +1,52 @@
...
-    }
-  };
+const Button: React.FC<ButtonProps> = ({
+	type = 'button',
+	disabled = false,
+	isLoading = false,
+	variant = 'primary',
+	children,
+}) => {
+	console.log(calculateDefaultDimensions(16, 50));
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:24]: **nitpick:** Remove log

File: src/inputs/Button/Button.tsx [hunk 5]
```diff
@@ -1,44 +1,52 @@
...
-  ].join(" ");
+	const allClasses = [globalButtonClass, variantClass, disabledClass, loadingClass].join(' ')
+	return (
+		<button className={allClasses} type={type} disabled={disabled}>
+			{isLoading ? (
+				<>
+					<span className={styles.iconSpinner}>
+						<Spinner />
+					</span>
+					{children}
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:38]: **nitpick:** This is still not formated

**Review by Sumick — CHANGES_REQUESTED** [review]
- [ ] Remove `package-lock.json`
- [ ] Refactor class names to use `cx` like in other components
- [ ] Fix formatting (it's still not properly formatted)

File: git reset --hard a0251b6c32f3167b7eb4a59fefb6067893ca7ca6 [hunk 6]
```diff
@@ -0,0 +1,226 @@
+[33mcommit 212692e65b07a0f8fef88b4350b982ba1df48701[m[33m ([m[1;36mHEAD -> [m[1;32mmain[m[33m)[m
```
Comment (Sumick) [inline git reset --hard a0251b6c32f3167b7eb4a59fefb6067893ca7ca6:1]: **nitpick**: We should delete this file

File: src/inputs/Button/Button.module.css [hunk 7]
```diff
@@ -1,58 +1,83 @@
...
 
-.template {
-  font-style: normal;
-  font-weight: 500;
-  background: #ffffff;
-  font-size: 14px;
-  min-width: 103px;
-  padding: 10px 24px;
-  line-height: 20px;
+.secondary:hover {
```
Comment (Sumick) [inline src/inputs/Button/Button.module.css:36]: **nitpick:** CSS is missing formatting here

File: src/inputs/Button/Button.tsx [hunk 8]
```diff
@@ -1,44 +1,55 @@
...
-  onClick(): void;
+interface ButtonProps {
+  type?: "button" | "submit";
   disabled?: boolean;
-  variant?: "primary" | "secondary";
-  isLoading?: boolean;
+  isLoading: boolean;
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  variant: "primary" | "secondary";
+  children: React.ReactNode;
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:15]: **nitpick:** We should not define children this way. Use `PropsWitchChildren<ButtonProps>` instead

**Review by Sumick — CHANGES_REQUESTED** [review]
- [ ] Remove redundant files (including `package-lock.json`)
- [ ] Fix component props interface
- [ ] Fix formatting for CSS

File: src/icons/Plus.tsx [hunk 9]
```diff
@@ -1,14 +1,23 @@
 import * as React from "react";
 
-export const Plus = () => (
+interface IconDefault {
+  color?: string;
+  scale?: number;
+}
+
+export const calculateDefaultDimensions = (x: number, scale = 100) => {
```
Comment (Sumick) [inline src/icons/Plus.tsx:8]: **nitpick:** This method should be somewhere else - maybe create `helpers.ts` file in `icons` directory, and put it there

File: src/inputs/Button/Button.stories.tsx [hunk 10]
```diff
@@ -1,23 +1,33 @@
...
+      },
     },
   },
 } as Meta;
 
-export const Button: Story = (args) => (
-  <MyButton {...(args as ButtonProps)}>Click me!</MyButton>
+export const Default: Story = (args) => (
+  <Button
+    icon={<Plus />}
```
Comment (Sumick) [inline src/inputs/Button/Button.stories.tsx:20]: **nitpick:** This icon looks definitely too big. Let's scale it down. + It does not look properly without icon (text is not centered):

Zrzut ekranu 2023-04-7 o 18 51 27

File: src/inputs/Button/Button.tsx [hunk 11]
```diff
@@ -1,44 +1,52 @@
...
-    styles.template,
-    variantClass,
-    getDisabledClassForSpecificVariant(),
-  ].join(" ");
+  const buttonClasses = cx({
+    buttons: true,
+    primary: variant === "primary",
+    secondary: variant === "secondary",
+    disabled: disabled,
+    isLoadingPrimary: variant === "primary" && isLoading && !disabled,
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:31]: **nitpick:** You duplicate these variant values - it's worth to move the to some `const` or `enum`, like:

`
enum ButtonVariant {
 primary = 'primary'
 ...
}
`

File: src/inputs/Button/Button.tsx [hunk 12]
```diff
@@ -1,44 +1,52 @@
...
+  });
 
   return (
-    <button
-      className={dynamicClasses}
-      onClick={props.onClick}
-      disabled={disabled}
-    >
-      {props.children}
+    <button className={buttonClasses} type={type} disabled={disabled}>
```
Comment (Sumick) [inline src/inputs/Button/Button.tsx:36]: **issue**: `isLoading` looks, and behaves different than on design. It should not have hover, and box shadow

Zrzut ekranu 2023-04-7 o 18 54 31

**Review by Sumick — CHANGES_REQUESTED** [review]
Almost there!

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
### Feedback
The code changes made in the previous messages are correct and meet the task requirements. However, there are some improvements that can be made to make the code more efficient and readable.

#### Improvements
- In `.husky/pre-commit` file, it is recommended to add comments explaining what each command does.
- In `src/icons/Plus.tsx` file, it is recommended to add a comment explaining what the `scaleCalculation` function does.
- In `src/inputs/Button/Button.module.css` file, it is recommended to remove unused CSS styles.
- In `src/inputs/Button/Button.stories.tsx` file, it is recommended to remove the `variant: "primary"` property from the `Default.args` object since it is already defined in the `<Button>` component.
- In `src/inputs/Button/Button.tsx` file, it is recommended to remove the duplicate declaration of `type`, `isLoading`, `variant`, `icon`, and `children`.
- It is also recommended to add comments explaining what each prop does in the `<Button>` component.

#### Task list
- [ ] Add comments explaining what each command does in `.husky/pre-commit`.
- [ ] Add a comment explaining what the `scaleCalculation` function does in `src/icons/Plus.tsx`.
- [ ] Remove unused CSS styles in `src/inputs/Button/Button.module.css`.
- [ ] Remove the `variant: "primary"` property from the `Default.args` object in `src/inputs/Button/Button.stories.tsx`.
- [ ] Remove duplicate declaration of props in `<Button>` component in

**Review by NerdbordMentor — APPROVED** [review]
### Feedback
The code changes made in the previous messages are correct and meet the task requirements. However, there are some improvements that can be made to make the code more efficient and readable.

#### Improvements
- In `.husky/pre-commit` file, it is recommended to add comments explaining what each command does.
- In `src/icons/Plus.tsx` file, it is recommended to add a comment explaining what the `scaleCalculation` function does.
- In `src/inputs/Button/Button.module.css` file, it is recommended to remove unused CSS styles.
- In `src/inputs/Button/Button.stories.tsx` file, it is recommended to remove the unnecessary `variant: "primary"` property from the `Default.args`.
- In `src/inputs/Button/Button.tsx` file, it is recommended to remove the unused import of `Fragment`.

#### Task List
- [ ] Add comments explaining what each command does in `.husky/pre-commit` file.
- [ ] Add a comment explaining what the `scaleCalculation` function does in `src/icons/Plus.tsx` file.
- [ ] Remove unused CSS styles in `src/inputs/Button/Button.module.css` file.
- [ ] Remove unnecessary `variant: "primary"` property from the `Default.args` in `src/inputs/Button/Button.stories.tsx` file.
- [ ] Remove unused import of `Fragment` in

**Review by NerdbordMentor — APPROVED** [review]
### Feedback
After reviewing the code changes, here is my feedback:
- The changes made to `.husky/pre-commit` file seem correct and meet the task requirements.
- The changes made to `src/icons/Plus.tsx` file seem correct and meet the task requirements.
- The changes made to `src/icons/scaleCalculation.ts` file seem correct and meet the task requirements.
- The changes made to `src/inputs/Button/Button.module.css` file seem correct and meet the task requirements.
- The changes made to `src/inputs/Button/Button.stories.tsx` file seem correct and meet the task requirements.
- The changes made to `src/inputs/Button/Button.tsx` file seem correct and meet the task requirements. However, there are some improvements that can be made:
  - [ ] Add a default value for the `variant` prop in case it is not provided by the user.
  - [ ] Remove redundant comments such as `// eslint-disable-next-line react/jsx-no-bind`.
  - [ ] Use destructuring assignment for props in function arguments instead of using `PropsWithChildren`.
  - [ ] Move the type definition of `ButtonProps` outside of the component function for better readability.
  - [ ] Use consistent naming conventions for CSS classes (e.g. use kebab-case instead of camelCase).
  
Overall, good job on implementing the Button component!

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**